src/odemis/driver/smaract.py

```diff
@@ -2931,6 +2955,15 @@ def _doReference(self, future, axes):
                         logging.warning("Axis %s not referenced after the end of referencing", a)
                         # TODO: Raise some error here
 
+                    # if referenced, move to the safe posiiton (if defined)
+                    if self._deactive_pos_after_ref is not None and self._deactive_pos_after_ref.get(a) is not None:
+                        deactive_pos = self._deactive_pos_after_ref[a]
+                        logging.info("Moving axis %s to deactivated position %f", a, deactive_pos)
+                        f = self._createMoveFuture()
+                        self._doMoveAbs(f, {a: deactive_pos})
```

Comment:
Probably you can also reuse the future here.

Reply:
Hmm.. when I tested it, it gets stuck here if you use the same future